test/tls_test.go

```diff
@@ -0,0 +1,175 @@
+		t.Error("Template should error")
+	}
+}
+
+func TestEnablePachTLSExistingSecret(t *testing.T) {
+	helmChartPath := "../pachyderm"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend": "LOCAL",
+			"pachd.tls.enabled":     "true",
+			"pachd.tls.secretName":  "blah",
+		},
+	}
+
+	manifest := helm.RenderTemplate(t, options, helmChartPath, "secret", []string{"templates/pachd/deployment.yaml"})
+	fmt.Println(manifest)
+}
+
+func TestEnableDashTLSNoName(t *testing.T) {
+	helmChartPath := "../pachyderm"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend": "LOCAL",
+			"dash.tls.enabled":      "true",
+			"dash.url":              "http://blah.com",
+		},
+	}
+
+	_, err := helm.RenderTemplateE(t, options, helmChartPath, "secret", []string{"templates/dash/ingress.yaml"})
+	if err == nil {
+		t.Error("Template should error")
+	}
+}
+
+func TestEnableDashTLSExistingSecret(t *testing.T) {
+	helmChartPath := "../pachyderm"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend": "LOCAL",
+			"dash.tls.enabled":      "true",
+			"dash.tls.secretName":   "blah",
+			"dash.url":              "http://blah.com",
+		},
+	}
+
+	manifest := helm.RenderTemplate(t, options, helmChartPath, "secret", []string{"templates/dash/ingress.yaml"})
+
+	fmt.Println(manifest)
+}
+
+func TestDashTLSNewSecretNoEnable(t *testing.T) {
+	helmChartPath := "../pachyderm"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend":  "LOCAL",
+			"dash.tls.newSecret.crt": "blah",
+			"dash.url":               "http://blah.com",
+		},
+	}
+
+	_, err := helm.RenderTemplateE(t, options, helmChartPath, "secret", []string{"templates/dash/ingress.yaml"})
+	if err == nil {
+		t.Error("Template should error")
+	}
+}
+
+func TestPachTLSNewSecretNoEnable(t *testing.T) {
+	helmChartPath := "../pachyderm"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend":   "LOCAL",
+			"pachd.tls.newSecret.crt": "blah",
+		},
+	}
+
+	_, err := helm.RenderTemplateE(t, options, helmChartPath, "deployment", []string{"templates/pachd/deployment.yaml"})
+	fmt.Println(err)
+	if err == nil {
+		t.Error("Template should error")
+	}
+}
+
+func TestPachTLSNewSecret(t *testing.T) {
+	helmChartPath := "../pachyderm"
+	secretName := "newsecret1234"
+	key := "blah"
+	crt := "blah"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend":      "LOCAL",
+			"pachd.tls.enabled":          "true",
+			"pachd.tls.newSecret.create": "true",
+			"pachd.tls.newSecret.crt":    crt,
+			"pachd.tls.newSecret.key":    key,
+			"pachd.tls.secretName":       secretName,
+		},
+	}
+	manifests := []string{
+		"templates/pachd/tls-secret.yaml",
+		"templates/pachd/deployment.yaml",
+	}
+
+	renderedManifest := helm.RenderTemplate(t, options, helmChartPath, "deployment", manifests)
+	objects, _ := manifestToObjects(renderedManifest)
```

Comment:
Check this error if that's the second return value.  It will save you some debugging; "manifest to objects failed: silly problem" is easier to debug than 0 objects being returned.  (You probably want to assert that you run through the v1.Secret and appsv1.Deployment cases, and fail if you don't.  Otherwise, 0 objects = passing test that should fail.)